src/pg/sql/40_observatory_utility.sql

```diff
@@ -0,0 +1,237 @@
+    SELECT tablename FROM observatory.OBS_table
+    WHERE id IN (
+      SELECT table_id
+      FROM observatory.OBS_table tab,
+           observatory.OBS_column_table coltable,
+           observatory.OBS_column col
+      WHERE type ILIKE ''geometry''
+        AND coltable.column_id = col.id
+        AND coltable.table_id = tab.id
+        AND col.id = $1
+        AND ST_Intersects($2, ST_SetSRID(bounds::box2d::geometry, 4326))
+    )
+    '
+  USING geometry_id, geom
+  INTO result;
+
+  return result;
+
+END;
+$$ LANGUAGE plpgsql;
+
+-- A type for use with the OBS_GetColumnData function
+CREATE TYPE OBS_ColumnData AS (colname text, tablename text, aggregate text);
+
+-- A function that gets teh column data for a column_id, geometry_id and timespan.
+-- Old: OBS_GetColumnData
+CREATE OR REPLACE FUNCTION OBS_GetColumnData(
+  geometry_id text,
+  column_id text,
+  timespan text
+)
+RETURNS OBS_ColumnData
+AS $$
+DECLARE
+  result OBS_ColumnData;
+BEGIN
+  EXECUTE '
+  WITH geomref AS (
+    SELECT t.table_id id
+    FROM observatory.OBS_column_to_column c2c, observatory.OBS_column_table t
+    WHERE c2c.reltype = ''geom_ref''
+      AND c2c.target_id = $1
+      AND c2c.source_id = t.column_id
+    )
+ SELECT colname, tablename, aggregate
+ FROM observatory.OBS_column c, observatory.OBS_column_table ct, observatory.OBS_table t
+ WHERE c.id = ct.column_id
+   AND t.id = ct.table_id
+   AND c.id = $2
+   AND t.timespan = $3
+   AND t.id in (SELECT id FROM geomref)
+ '
+ USING geometry_id, column_id, timespan
+ INTO result;
+ RETURN result;
+
+END;
+$$ LANGUAGE plpgsql;
+
+
+-- A function that gets the column data for multiple columns
+-- Old: OBS_GetColumnData
+CREATE OR REPLACE FUNCTION OBS_GetColumnData(
+  geometry_id text,
+  column_ids text[],
+  timespan text
+)
+RETURNS OBS_ColumnData[]
+AS $$
+DECLARE
+  result OBS_ColumnData[];
+BEGIN
+
+  EXECUTE '
+  WITH geomref AS (
+    SELECT t.table_id id
+    FROM observatory.OBS_column_to_column c2c, observatory.OBS_column_table t
+    WHERE c2c.reltype = ''geom_ref''
+      AND c2c.target_id = $1
+      AND c2c.source_id = t.column_id
+    ),
+  column_ids as (
+    select row_number() over () as no, a.column_id as column_id from (select unnest($2) as column_id) a
+  )
+ SELECT array_agg(ROW(colname, tablename, aggregate)::OBS_ColumnData order by column_ids.no)
+ FROM column_ids, observatory.OBS_column c, observatory.OBS_column_table ct, observatory.OBS_table t
+ WHERE column_ids.column_id  = c.id
+   AND c.id = ct.column_id
+   AND t.id = ct.table_id
+   AND t.timespan = $3
+   AND t.id in (SELECT id FROM geomref)
+ '
+ USING geometry_id, column_ids, timespan
+ INTO result;
+ RETURN result;
+
+END;
+$$ LANGUAGE plpgsql;
+
+--Gets the column id for a census variable given a human readable version of it
+-- Old: OBS_LOOKUP_CENSUS_HUMAN
+CREATE OR REPLACE FUNCTION OBS_LookupCensusHuman(
+  column_name text,
+  table_name text DEFAULT '"us.census.acs".extract_block_group_5yr_2013_69b156927c'
```

Comment:
It's a hash generated by Luigi based off of task parameters.

There's a TODO around defaulting a tablename like this -- we should be looking up the tablename from metadata instead of hardcoding it.  That would insulate us against possible changes in the hash.